snapcraft/tests/unit/test_errors.py

```diff
@@ -160,21 +160,20 @@ class ErrorFormattingTestCase(unit.TestCase):
                 "Use a LXD remote without colons, spaces and slashes in the "
                 "name.\n")}),
 
-        ('InvalidDesktopFileError', {
-            'exception': errors.InvalidDesktopFileError,
+        ('', {
+            'exception': errors.MissingDesktopFileError,
             'kwargs': {
-                'filename': 'test-file',
-                'message': 'test-message'
+                'filename': 'test-file'
```

Comment:
You need the app_name here as well. Then the message below should use them, not just be a copy/paste of the string from the exception, or the test won't pass.